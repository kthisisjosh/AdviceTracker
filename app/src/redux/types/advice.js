export const GET_INBOX_ADVICE = "GET_INBOX_ADVICE"
export const SUBMIT_INBOX_ADVICE = "SUBMIT_INBOX_ADVICE"
export const DELETE_INBOX_ADVICE = "DELETE_INBOX_ADVICE"
export const GET_ADVICE = "GET_ADVICE"
export const SUBMIT_CATEGORY = "SUBMIT_CATEGORY"
