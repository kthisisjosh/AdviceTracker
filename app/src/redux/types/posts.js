export const CREATE_POST = "CREATE_POST"
export const GET_POST = "GET_POST"
