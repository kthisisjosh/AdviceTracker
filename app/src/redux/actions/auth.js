import { LOGIN_SUCCESS, LOGIN_FAIL } from "../types/auth"
import { sessionService } from "redux-react-session"
import { submitAdvice, submitCategory, submitSubCategory, getAdvice } from "./advice"
import Swal from "sweetalert2"

const Toast = Swal.mixin({
    toast: true,
    position: "top-end",
    showConfirmButton: false,
    timer: 2500,
    onOpen: (toast) => {
        toast.addEventListener("mouseenter", Swal.stopTimer)
        toast.addEventListener("mouseleave", Swal.resumeTimer)
    },
})

export const googleLogin = (response, history, categories) => async (dispatch) => {
    const profileObj = response.profileObj
    try {
        const newUser = {
            id: profileObj.googleId,
            username: profileObj.name,
            email: profileObj.email,
            profileUrl: profileObj.imageUrl,
        }
        const options = {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(newUser),
        }
        const url = "https://advicetracker.life/api/users/"
        await fetch(url, options)
            .then((response) => response.json())
            .then((data) => {
                sessionService
                    .saveSession({ token: data.token })
                    .then(() => {
                        sessionService
                            .saveUser(data.user)
                            .then(() => {
                                dispatch({ type: LOGIN_SUCCESS, payload: { ...data, token: data.token } })
                                history.push("/dashboard")
                                Toast.fire({
                                    icon: "success",
                                    title: "Signed in successfully",
                                })
                            })
                            .catch((err) => console.log(err))
                    })
                    .catch((err) => console.log(err))
            })
            .catch(() => {
                dispatch({ type: LOGIN_FAIL })
            })
    } catch (error) {
        dispatch({ type: LOGIN_FAIL })
    }
}

export const githubLogin = ({ profileObj }) => async (dispatch) => {
    //
}
