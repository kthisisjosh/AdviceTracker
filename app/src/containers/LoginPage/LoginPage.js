import React, { useEffect } from "react"
import { googleLogin, githubLogin } from "../../redux/actions/auth"
import { connect } from "react-redux"
import Typography from "@material-ui/core/Typography"
import LandingHeader from "../../components/LandingHeader/LandingHeader"
import GoogleLogin from "react-google-login"
import GithubLogin from "react-github-login"
import Button from "@material-ui/core/Button"
import Footer from "../../components/Footer/Footer"
import Paper from "@material-ui/core/Paper"
import Grid from "@material-ui/core/Grid"
import { useHistory } from "react-router-dom"
import { Helmet } from "react-helmet"

const LoginPage = (props) => {
    const { googleLogin, githubLogin, isAuthenticated, user } = props
    const history = useHistory()

    useEffect(() => {
        if (isAuthenticated) {
            history.push("/dashboard")
        } else {
        }
    }, [isAuthenticated, user, history])

    return (
        <>
            <Helmet>
                <title>Login | AdviceTracker</title>
                <meta
                    name="description"
                    content="Login today to track the awesome advice you are given, so you are always one step ahead. Find & save valuable advice on the way by browsing
                    user submitted advice."
                />
                <link rel="canonical" href="https://advicetracker.life/login"/>
            </Helmet>
            <LandingHeader />
            <section>
                <Grid container style={{ height: "750px" }}>
                    <Grid
                        item
                        className="login-demo-grid"
                        xl={5}
                        md={5}
                        style={{
                            height: "750px",
                            backgroundColor: "#EDEDED",
                            borderTop: "1px solid black",
                            borderBottom: "1px solid black",
                            borderRight: "1px solid black",
                            textAlign: "center",
                        }}
                    >
                        <img
                            alt="login-demo"
                            src="./login-sample.JPG"
                            style={{ height: "550px", width: "500px", margin: "75px 15px 35px 15px", borderRadius: "15px" }}
                        />
                    </Grid>
                    <Grid
                        item
                        className="login-grid"
                        xl={7}
                        md={7}
                        sm={12}
                        xs={12}
                        style={{
                            height: "750px",
                            backgroundColor: "#FCFCFC",
                            borderTop: "1px solid black",
                            borderBottom: "1px solid black",
                            padding: "auto",
                        }}
                    >
                        <Paper
                            className="login-paper"
                            elevation={3}
                            style={{ margin: "70px 350px 50px 150px", position: "absolute", width: "460px", height: "500px" }}
                        >
                            <div style={{ width: "460px, height: 200px" }}>
                                <Button style={{ marginTop: "50px" }}>Registration</Button>
                                <Button style={{ marginTop: "50px" }}>Sign-in</Button>
                            </div>
                            <div style={{ position: "relative", width: "460px", height: "50px", backgroundColor: "#FAFAFA", alignText: "center" }}>
                                <Typography style={{ paddingLeft: "8px", paddingTop: "12.5px" }}>Sign in with social</Typography>
                            </div>
                            <div style={{ textAlign: "center", height: "300px", backgroundColor: "#FEFEFE" }}>
                                <div style={{ display: "inline-block", paddingTop: "50px" }}>
                                    <GoogleLogin
                                        clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
                                        onSuccess={googleLogin}
                                        onFailure={googleLogin}
                                        disabled={false}
                                    />
                                </div>
                                <div style={{ paddingTop: "50px" }}></div>
                                <div style={{ paddingTop: "75px" }}>
                                    <Button style={{ width: "350px", height: "50px", backgroundColor: "#F2994A", color: "#FFFFFF" }}>
                                        Register Coming Soon!
                                    </Button>
                                </div>
                            </div>
                        </Paper>
                    </Grid>
                </Grid>
            </section>
            <Footer />
        </>
    )
}

const mapStateToProps = (state) => ({
    isAuthenticated: state.authState.isAuthenticated,
    user: state.authState.user,
})

export default connect(mapStateToProps, { googleLogin, githubLogin })(LoginPage)

//<GithubLogin
//
//    buttonText="Sign in using Github"
//    onSuccess={githubLogin}
//    onFailure={githubLogin}
//    className="github-login-btn"
///>
