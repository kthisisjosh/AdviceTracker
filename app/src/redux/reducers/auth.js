import { LOGIN_SUCCESS, LOGIN_FAIL, LOGOUT } from "../types/auth"

const initialState = {
    token: localStorage.getItem("jwtToken"),
    isAuthenticated: null,
    loading: true,
    user: null,
}

export default (state = initialState, action) => {
    switch (action.type) {
        case LOGIN_SUCCESS:
            localStorage.setItem("jwtToken", action.payload.token)
            return { ...state, isAuthenticated: true, loading: false, user: action.payload.user }
        case LOGIN_FAIL:
        case LOGOUT:
            localStorage.removeItem("jwtToken")
            return initialState
        default:
            return state
    }
}
