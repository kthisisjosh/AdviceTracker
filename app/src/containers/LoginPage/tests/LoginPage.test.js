import React from "react"
import store from "../../../redux"
import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import LoginPage from "../LoginPage"
import { BrowserRouter as Router } from "react-router-dom"
import authReducer from "../../../redux/reducers/auth"
import { LOGIN_SUCCESS, LOGOUT } from "../../../redux/types/auth"

describe("<LoginPage />", () => {
    it("should render and match the snapshot", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <LoginPage />
                </Router>
            </Provider>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    describe("LoginPage Auth Reducer", () => {
        let state
        beforeEach(() => {
            state = {
                token: null,
                isAuthenticated: null,
                loading: true,
                user: null,
            }
        })

        it("should return the initial state", () => {
            const expectedResult = state
            expect(authReducer(undefined, {})).toEqual(expectedResult)
        })

        it("should handle the LOGIN_SUCCESS dispatch correctly", () => {
            const fixture = {
                token: "test-token",
                user: {
                    userID: "test-id",
                    username: "test",
                    email: "test",
                    token: "test-token",
                },
            }
            const expectedResult = {
                token: null,
                isAuthenticated: true,
                loading: false,
                user: {
                    userID: "test-id",
                    username: "test",
                    email: "test",
                    token: "test-token",
                },
            }

            expect(authReducer(state, { type: LOGIN_SUCCESS, payload: fixture })).toEqual(expectedResult)
        })

        it("should handle the LOGOUT dispatch correctly", () => {
            const loginState = {
                token: "test-token",
                user: {
                    userID: "test-id",
                    username: "test",
                    email: "test",
                    token: "test-token",
                },
            }

            expect(authReducer(loginState, { type: LOGOUT })).toEqual(state)
        })
    })
})
