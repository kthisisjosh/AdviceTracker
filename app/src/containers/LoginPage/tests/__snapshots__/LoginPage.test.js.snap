// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render and match the snapshot 1`] = `
Array [
  <header
    id="header"
    style={
      Object {
        "margin": "0",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style={
        Object {
          "alignContent": "center",
          "height": "60px",
        }
      }
    >
      <div
        className="MuiGrid-root landing-logo MuiGrid-item"
        style={
          Object {
            "marginLeft": "14vw",
            "marginTop": "1vh",
          }
        }
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            src="./AdviceLogoLight.png"
          />
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
        style={
          Object {
            "marginLeft": "auto",
            "marginRight": "5vw",
          }
        }
      >
        <div
          className="MuiGrid-root toggle-grid MuiGrid-item MuiGrid-grid-md-6"
        >
          <div
            className="react-toggle react-toggle--checked react-toggle-small"
            onClick={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="react-toggle-track"
            >
              <div
                className="react-toggle-track-check"
              >
                <img
                  alt="moon"
                  aria-hidden={true}
                  height="14"
                  src="./moon.png"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  width="16"
                />
              </div>
              <div
                className="react-toggle-track-x"
              >
                <img
                  alt="sun"
                  aria-hidden={true}
                  height="14"
                  src="./sun.png"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  width="16"
                />
              </div>
            </div>
            <div
              className="react-toggle-thumb"
            />
            <input
              className="react-toggle-screenreader-only"
              defaultChecked={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
        >
          <div>
            <div
              className="bm-menu-wrap"
              id=""
              style={
                Object {
                  "MozTransform": "translate3d(100%, 0, 0)",
                  "MsTransform": "translate3d(100%, 0, 0)",
                  "OTransform": "translate3d(100%, 0, 0)",
                  "WebkitTransform": "translate3d(100%, 0, 0)",
                  "height": "100%",
                  "position": "fixed",
                  "right": 0,
                  "transform": "translate3d(100%, 0, 0)",
                  "transition": "all 0.5s",
                  "width": "300px",
                  "zIndex": 1100,
                }
              }
            >
              <div
                className="bm-menu"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "height": "100%",
                    "overflow": "auto",
                  }
                }
              >
                <nav
                  className="bm-item-list"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <a
                    className="bm-item"
                    href="/browse"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "white",
                        "display": "block",
                        "marginBottom": "2vw",
                        "textDecoration": "none",
                      }
                    }
                    tabIndex={-1}
                  >
                    <span
                      className="MuiTypography-root MuiTypography-button"
                    >
                      Browse
                    </span>
                  </a>
                  <a
                    className="bm-item"
                    href="/login"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "white",
                        "display": "block",
                        "textDecoration": "none",
                      }
                    }
                    tabIndex={-1}
                  >
                    <span
                      className="MuiTypography-root MuiTypography-button"
                    >
                      Login/Register
                    </span>
                  </a>
                </nav>
              </div>
              <div
                style={Object {}}
              >
                <div
                  className="bm-cross-button"
                  style={
                    Object {
                      "height": 24,
                      "position": "absolute",
                      "right": 8,
                      "top": 8,
                      "width": 24,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "position": "absolute",
                        "right": "14px",
                        "top": "6px",
                      }
                    }
                  >
                    <span
                      className="bm-cross"
                      style={
                        Object {
                          "height": 14,
                          "position": "absolute",
                          "transform": "rotate(45deg)",
                          "width": 3,
                        }
                      }
                    />
                    <span
                      className="bm-cross"
                      style={
                        Object {
                          "height": 14,
                          "position": "absolute",
                          "transform": "rotate(-45deg)",
                          "width": 3,
                        }
                      }
                    />
                  </span>
                  <button
                    onClick={[Function]}
                    style={
                      Object {
                        "background": "transparent",
                        "border": "none",
                        "cursor": "pointer",
                        "fontSize": 0,
                        "height": "100%",
                        "left": 0,
                        "margin": 0,
                        "padding": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    Close Menu
                  </button>
                </div>
              </div>
            </div>
            <div
              style={Object {}}
            >
              <div
                className="bm-burger-button"
                style={
                  Object {
                    "zIndex": 1000,
                  }
                }
              >
                <span>
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "0%",
                      }
                    }
                  />
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "40%",
                      }
                    }
                  />
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "80%",
                      }
                    }
                  />
                </span>
                <button
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "background": "transparent",
                      "border": "none",
                      "cursor": "pointer",
                      "fontSize": 0,
                      "height": "100%",
                      "left": 0,
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                >
                  Open Menu
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root navbar-grid MuiGrid-item"
        style={
          Object {
            "marginLeft": "auto",
            "textAlign": "center",
          }
        }
      >
        <div
          className="react-toggle react-toggle--checked"
          onClick={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="react-toggle-track"
          >
            <div
              className="react-toggle-track-check"
            >
              <img
                alt="moon"
                aria-hidden={true}
                height="14"
                src="./moon.png"
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                width="16"
              />
            </div>
            <div
              className="react-toggle-track-x"
            >
              <img
                alt="sun"
                aria-hidden={true}
                height="14"
                src="./sun.png"
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                width="16"
              />
            </div>
          </div>
          <div
            className="react-toggle-thumb"
          />
          <input
            className="react-toggle-screenreader-only"
            defaultChecked={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="checkbox"
          />
        </div>
        <a
          className="landing-header-link"
          href="/browse"
          onClick={[Function]}
          style={
            Object {
              "margin": "0 0 0 5vw",
              "textDecoration": "none",
            }
          }
        >
          <span
            className="MuiTypography-root MuiTypography-button"
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            Browse
          </span>
        </a>
      </div>
      <div
        className="MuiGrid-root navbar-grid MuiGrid-item"
        style={
          Object {
            "marginLeft": "5vw",
            "marginRight": "1vw",
          }
        }
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined landing-hero-btn"
          href="/login"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#f5f5f5",
              "color": "#2D9CDB",
              "height": "40px",
              "width": "130px",
            }
          }
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
              Login
            </span>
          </span>
        </a>
      </div>
      <div
        className="MuiGrid-root navbar-grid MuiGrid-item"
        style={
          Object {
            "marginLeft": "1vw",
            "marginRight": "15vw",
          }
        }
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          href="/login"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#F2994A",
              "color": "#FFFFFF",
              "height": "40px",
              "width": "130px",
            }
          }
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
              Register
            </span>
          </span>
        </a>
      </div>
    </div>
  </header>,
  <section>
    <div
      className="MuiGrid-root MuiGrid-container"
      style={
        Object {
          "height": "750px",
        }
      }
    >
      <div
        className="MuiGrid-root login-demo-grid MuiGrid-item MuiGrid-grid-md-5 MuiGrid-grid-xl-5"
        style={
          Object {
            "backgroundColor": "#EDEDED",
            "borderBottom": "1px solid black",
            "borderRight": "1px solid black",
            "borderTop": "1px solid black",
            "height": "750px",
            "textAlign": "center",
          }
        }
      >
        <img
          alt="login-demo"
          src="./login-sample.JPG"
          style={
            Object {
              "borderRadius": "15px",
              "height": "550px",
              "margin": "75px 15px 35px 15px",
              "width": "500px",
            }
          }
        />
      </div>
      <div
        className="MuiGrid-root login-grid MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-7 MuiGrid-grid-xl-7"
        style={
          Object {
            "backgroundColor": "#FCFCFC",
            "borderBottom": "1px solid black",
            "borderTop": "1px solid black",
            "height": "750px",
            "padding": "auto",
          }
        }
      >
        <div
          className="MuiPaper-root login-paper MuiPaper-elevation3 MuiPaper-rounded"
          style={
            Object {
              "height": "500px",
              "margin": "70px 350px 50px 150px",
              "position": "absolute",
              "width": "460px",
            }
          }
        >
          <div
            style={
              Object {
                "width": "460px, height: 200px",
              }
            }
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "marginTop": "50px",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Registration
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "marginTop": "50px",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign-in
              </span>
            </button>
          </div>
          <div
            style={
              Object {
                "alignText": "center",
                "backgroundColor": "#FAFAFA",
                "height": "50px",
                "position": "relative",
                "width": "460px",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
              style={
                Object {
                  "paddingLeft": "8px",
                  "paddingTop": "12.5px",
                }
              }
            >
              Sign in with social
            </p>
          </div>
          <div
            style={
              Object {
                "backgroundColor": "#FEFEFE",
                "height": "300px",
                "textAlign": "center",
              }
            }
          >
            <div
              style={
                Object {
                  "display": "inline-block",
                  "paddingTop": "50px",
                }
              }
            >
              <button
                disabled={true}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "border": "1px solid transparent",
                    "borderRadius": 2,
                    "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                    "color": "rgba(0, 0, 0, .54)",
                    "display": "inline-flex",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "opacity": 0.6,
                    "padding": 0,
                  }
                }
                type="button"
              >
                <div
                  style={
                    Object {
                      "background": "#fff",
                      "borderRadius": 2,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <svg
                    height="18"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#000"
                      fillRule="evenodd"
                    >
                      <path
                        d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                        fill="#EA4335"
                      />
                      <path
                        d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                        fill="#4285F4"
                      />
                      <path
                        d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                        fill="#FBBC05"
                      />
                      <path
                        d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                        fill="#34A853"
                      />
                      <path
                        d="M0 0h18v18H0z"
                        fill="none"
                      />
                    </g>
                  </svg>
                </div>
                <span
                  style={
                    Object {
                      "fontWeight": 500,
                      "paddingBottom": 10,
                      "paddingLeft": 0,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    }
                  }
                >
                  Sign in with Google
                </span>
              </button>
            </div>
            <div
              style={
                Object {
                  "paddingTop": "50px",
                }
              }
            />
            <div
              style={
                Object {
                  "paddingTop": "75px",
                }
              }
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#F2994A",
                    "color": "#FFFFFF",
                    "height": "50px",
                    "width": "350px",
                  }
                }
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Register Coming Soon
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "auto",
        "margin": "0",
        "right": "0",
        "textAlign": "center",
        "width": "100vw",
      }
    }
  >
    <div
      style={
        Object {
          "marginTop": "5vh",
        }
      }
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          src="./AdviceLogo.png"
        />
      </a>
    </div>
    <hr
      style={
        Object {
          "width": "auto",
        }
      }
    />
    <p
      className="MuiTypography-root MuiTypography-body2"
      style={
        Object {
          "lineHeight": "24px",
          "margin": "auto",
          "marginTop": "35px",
        }
      }
    >
      © Advice Tracker 2020
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
      style={
        Object {
          "lineHeight": "24px",
          "margin": "auto",
        }
      }
    >
      Made with
       
      <span
        aria-label="Heart"
        role="img"
      >
        ❤️
      </span>
       
      by
       
      <a
        href="https://www.joshbautista.com"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        Joshua Bautista
      </a>
    </p>
  </footer>,
]
`;
